### PR TITLE
Adds plasma gun and energy gunmods

### PR DESCRIPTION
**Pull request by chaisawlajatang**
Adds refluffed and modified plasma gun. Discussion is [here](https://discourse.cataclysmdda.org/t/viability-of-plasma-weaponry/20794/2).

Volume and weight are based on in-game heavy weaponry (namely, heavy railgun), most of other parameters are based on old FTK and fusion rifle (the one made from the CBM). I replaced EXPLOSIVE flag with EXPLOSIVE_SMALL. MARAUDER project claims to be able to create devastating plasma explosions, but I decided to reduce destructive potential a bit because I guess we should have a larger scale weapon for that (maybe turret-sized or 20-kg gun?) and for balance reasons. I don't want to make this overpowered.
I added EMP damage, because sources say plasma cannon IRL would do such a thing. 

So the niche for this weapon would be dealing with mid- and lategame armored enemies and robots. 

Also changed vehicle turret accordingly.

#### Summary

```SUMMARY: Content "Adds refluffed and modified plasma gun"```

#### Purpose of change

Because plasma hand-held weaponry is realistic and can be extrapolated from our current tech level, especially with some advanced technology we have in Cataclysm.

#### Describe the solution

Add plasma gun.

#### Describe alternatives you've considered

Don't add plasma gun.

#### Additional context

I thought about making plasma gun a purely anti-robo weapon, but I suddenly discovered we actually have https://github.com/CleverRaven/Cataclysm-DDA/pull/15510 this.

There was almost never-spawning group of energy gunmods, I made them actually spawn and added a couple of extra gunmods for variety sake. 

Also added racks to military bunker and slighlty upped spawnrate of hydrogen tanks, energy guns and related stuff. It's still very-very rare.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
> I thought about making plasma gun a purely anti-robo weapon, but I suddenly discovered we actually have #15510 this.

Of note, this gun is rare enough most people haven't seen it.
You may also want to check out the XM-P, an already existing plasma gun (that I didn't realize we had).

Anyways, thanks for doing this!

File: data/json/itemgroups/guns.json [hunk 1]
```diff
@@ -224,7 +224,7 @@
     "type": "item_group",
     "id": "guns_energy",
     "//": "Assorted factory crafted energy weapons.",
-    "items": [ { "item": "laser_rifle", "prob": 30 }, { "item": "emp_gun", "prob": 10 }, { "item": "v29", "prob": 70 } ]
+    "items": [ { "item": "laser_rifle", "prob": 30 }, { "item": "emp_gun", "prob": 10 }, { "item": "v29", "prob": 70 }, { "item": "plasma_gun", "prob": 10 } ]
```
Comment (anothersimulacrum) [inline data/json/itemgroups/guns.json]: Your this file needs to be linted, as such.
```suggestion
    "items": [ 
      { "item": "laser_rifle", "prob": 30 },
      { "item": "emp_gun", "prob": 10 },
      { "item": "v29", "prob": 70 },
      { "item": "plasma_gun", "prob": 10 }
    ]
```

**Comment by Mecares** [comment]
The plasma gun should use some kind of ammo.

http://hyperv.com/pubs/RSINAK808083506_1.pdf

In this paper a thin layer of polyethylen that is vaporized is used to create the plasma is mentioned.

File: data/json/items/ranged.json [hunk 2]
```diff
@@ -222,6 +222,41 @@
...
+    "name": "PPA-5",
+    "description": "Portable Plasma Accelerator Model Five, developed by Lockheed Martin Corporation. This device uses highly advanced capacitor banks to create a torus of superheated plasma and accelerate it to incredible speed. It was designed to take down heavy vehicles, and was expected to fully enter US Army service not long before the Cataclysm.",
+    "price": 980000,
+    "material": [ "steel", "plastic" ],
+    "ammo_effects": [ "PLASMA", "FLAME", "EMP", "EXPLOSIVE_SMALL" ],
+    "flags": [ "NEVER_JAMS", "NO_UNLOAD" ],
+    "ups_charges": 100,
+    "skill": "rifle",
+    "weight": 10500,
+    "volume": 16,
```
Comment (anothersimulacrum) [inline data/json/items/ranged.json]:
```suggestion
    "volume": "4L",
```

**Comment by chaisawlajatang** [comment]
>The plasma gun should use some kind of ammo.

I thought about it and not completely sure. I made this gun to run solely on UPS for several reasons.

Looks like that device uses a very tiny bit of matter to create plasma - solid polyethylene covers an electrode and vaporizes layer by layer - and resulting plasma is measured in micrograms. I suppose military-grade device could create more plasma (maybe even milligrams of it), but the resulting factors are temperature and velocity, not the mass. 

And more importantly, authors say (page 3):

> Ultimately the ablative [polyethylene] walls will be replaced with nonablative refractory ceramics, as has been accomplished for pulsed soft x-ray sources and for high energy pulsed thermal spray devices. This will allow clean and repetitive operation. The H2 or D2 or other, higher-Z gases in this case are introduced by other means, which are currently under development at HyperV and will be addressed in future papers.

**Comment by kevingranade** [comment]
That seems to indicate that they're switching from an ablative source to a
gaseous source, I don't see anything indicating use of ambient air.

**Comment by I-am-Erk** [comment]
Quite the opposite, sounds like it's running on compressed hydrogen or deuterium. Makes sense, I don't think it would be even remotely safe to try running something like that on room air.

**Comment by chaisawlajatang** [comment]
Ok, that's right. I added hydrogen as ammo, a canister would spawn near the gun, I mentioned hydrogen in the description. 
I'm preparing another PR with some spawn rebalancing, so hydrogen (and some other stuff) would not be so prohibitely rare.

**Comment by chaisawlajatang** [comment]
Hmmm, IDK why the checks are failed.

**Comment by Night-Pryanik** [comment]
изображение

изображение

изображение

Which means you have to [lint](http://dev.narc.ro/cataclysm/format.html) your json changes.

**Comment by chaisawlajatang** [comment]
Energy gunmods are so rare I never saw one in my whole playtime. I added them in a separate group along with some extra energy gunmods. Apparently we had almost never-spawning group energy_weapon_armory, so I included all the new stuff there.

**Comment by chaisawlajatang** [comment]
In general, energy weapons and mods would spawn in lab armories and military outposts (I added racks to underground bunker, overall chance to find any item is roughly the same as before) with a pretty low chance. They are still extremely rare, but finally have a chance to actually spawn (not counting hardcoded places like lab finale).

**Comment by Night-Pryanik** [comment]
Jenkins rebuild

**Pull request merged**